### ISSUE TITLE
Password Protected Modules Not Displaying in RubberDuck Project Exporer

### ISSUE THREAD

**Issue by tallguyjenks** [rubberduck-vba/Rubberduck#4710]

**Comment by bclothier** [comment]
AIUI, that is going to be fixed in upcoming PR #4535

**Comment by retailcoder** [comment]
This *was* by-design, since the VBIDE API doesn't allow us to access the modues of protected projects. With the linked PR, we'll be able to list the members and use them in the identifier resolution logic.

**Comment by tallguyjenks** [comment]
shoot, I fat fingered the enter button and it shot off pre-maturely.

I downloaded the latest version 2 minutes ago to be sure this wasn't resolved and searched open issues, but when I have a password protected VBA module it doesn't appear in the code explorer of Rubber duck and if I want to unlock the file for editing I need to go through the legacy project explorer menu. not sure if its intentional but I lock my projects for distribution in my organization and not having them appear in the code explorer is an inconvenience when I want to fuller use Rubber Duck in the IDE, screen shot attached

**Comment by tallguyjenks** [comment]
@retailcoder so this will eventually be changed to display even the protected modules?

**Comment by retailcoder** [comment]
Kind of - we'll be able to list the protected modules and their members (and signatures), but we can't override the VBE's protection and display the source code / navigate to locked projects' modules.

I gather that having a toggle-button to unlock a locked project would be a nice-to-have addition?

**Comment by tallguyjenks** [comment]
im mainly just look to SEE the protected module in the RubberDuck Code explorer and be able to click on it and be prompted for the unlock password just like the original VBIDE, because I have the password for my protected modules, they're just not even displaying in the code explorer

**Comment by retailcoder** [comment]
Yup, that's the plan! Note that there's very non-trivial black magic voodoo going on for this to happen - if you try to access the properties of a locked project's `VBComponent`, the VBIDE API throws an error (a COM exception, if you're doing this in .net).

**Comment by tallguyjenks** [comment]
okay sweet! then I look forward to that! :)
